python-regression/util/test_logic/api_test_logic.py

```diff
@@ -84,33 +84,49 @@ def prepare_options(args, option_list):
 
             if arg_type == "int":
                 value = int(value)
+
             elif arg_type == "list":
                 value = [value]
+
             elif arg_type == "nodeAddress":
                 host = world.machine['nodes'][value]['host']
                 port = world.machine['nodes'][value]['ports']['gossip-udp']
                 address = "udp://" + host + ":" + str(port)
                 value = [address.decode()]
+
             elif arg_type == "staticValue":
                 value = getattr(static_vals, value)
+
             elif arg_type == "staticList":
                 address = getattr(static_vals, value)
                 value = [address]
+
             elif arg_type == "bool":
                 if value == "False":
                     value = False
                 else:
                     value = True
+
             elif arg_type == "responseValue":
                 config = fetch_config('nodeId')
                 response = fetch_response(value)
                 value = response[config]
+
             elif arg_type == "responseList":
                 config = fetch_config('nodeId')
                 response = fetch_response(value)
                 value = [response[config]]
 
-            option_list[key] = value
+            elif arg_type == "configValue":
+                node = fetch_config('nodeId')
+                value = world.config[value][node]
+
+            elif arg_type == "configList":
+                node = fetch_config('nodeId')
+                value = [world.config[value][node]]
+
+            if key != 'seed':
+                option_list[key] = value
```

Comment:
This will grow big in time,
we need to think about it

Reply:
Yeah I agree, it'll grow too big too quickly, so i'll be separating the if/elifs into function calls and do a dictionary call list the same way I prepared the api calls. This will make it more readable and will make expanding it much easier.